src/components/header/HeaderItem.tsx

```diff
@@ -0,0 +1,37 @@
+import { ListItemButton, ListItemIcon, ListItemText } from '@mui/material';
+import ListItem from '@mui/material/ListItem';
+
+const HeaderItem = ({
+  isHeaderOpen,
+  icon,
+  text,
+}: {
```

Comment:
@BardhMustafa every component that has props should have the type declared on top, the solution here would be crowded if we add more props, I suggest creating a HeaderItemProp type